esp/services/esdl_svc_engine/esdl_svc_custom.cpp

```diff
@@ -306,13 +306,46 @@ void CEsdlCustomTransform::processTransform(IEspContext * context, StringBuffer
         auto user = context->queryUser();
         if (user)
         {
+            static const std::map<SecUserStatus, const char*> statusLabels =
+            {
+#define STATUS_LABEL_NODE(s) { s, #s }
+                STATUS_LABEL_NODE(SecUserStatus_Inhouse),
+                STATUS_LABEL_NODE(SecUserStatus_Active),
+                STATUS_LABEL_NODE(SecUserStatus_Exempt),
+                STATUS_LABEL_NODE(SecUserStatus_FreeTrial),
+                STATUS_LABEL_NODE(SecUserStatus_FreeTrial),
+                STATUS_LABEL_NODE(SecUserStatus_Rollover),
+                STATUS_LABEL_NODE(SecUserStatus_Suspended),
+                STATUS_LABEL_NODE(SecUserStatus_Terminated),
+                STATUS_LABEL_NODE(SecUserStatus_TrialExpired),
+                STATUS_LABEL_NODE(SecUserStatus_Status_Hold),
+                STATUS_LABEL_NODE(SecUserStatus_Unknown),
```

Comment:
SecUserStatus_Unknown is the default user status. Various platform files specifically test for this status, suggesting there is an expectation that it will be unchanged. I think the transform should be able to see the status and decide for itself what it means.

Reply:
the logic using this table should report SecUserStatus_Unknown, it doesn't make sense to have the error/default status in the look up table.
 auto it = statusLabels.find(user->getStatus());
if (it != statusLabels.end()) 
                xpathContext->addVariable("secuser_status_string", it->second); //how do you know this wasn't SecUserStatus_Unknown ?
            else
                xpathContext->addVariable("secuser_status_string", "unexpected");

Reply:
A better name for SecUserStatus_Unknown might be SecUserStatus_Default. This is the default value for all ISecUser instances, and the only value returned by the LDAP secure user. The value must remain in the map as an expected value, since unexpected values will now trigger exceptions.